test/controllers/auth_controller_test.rb

```diff
@@ -2,15 +2,19 @@
 
 require 'test_helper'
 
-class HealthCardsControllerTest < ActionDispatch::IntegrationTest
-  test 'authorize without client_id should return 401' do
+class AuthControllerTest < ActionDispatch::IntegrationTest
+  test 'authorize without client_id should return 400 invalid_request' do
     get(auth_code_path)
-    assert_response :unauthorized
+    assert_response :bad_request
+    assert_no_cache_headers
+    assert_equal('{"error":"invalid_request"}', @response.body)
   end
 
-  test 'authorize with incorrect client_id should return 401' do
-    get(auth_code_path, params: { client_id: 'id' })
-    assert_response :unauthorized
+  test 'authorize with incorrect client_id should return 400 invalid_client' do
```

Comment:
Is there ever a situation where this will return a 401 instead of a 400, according to https://datatracker.ietf.org/doc/html/rfc6749#section-5.2?

Reply:
I think that should be in the SecuredController, when client sends an Authorize http header but the bearer token is incorrect a 401 must be returned. In the AuthController the client shouldn't be sending any Authorization header, just a GET /auth/authorize?client_id=something.

Reply:
But in SecuredController right now we just accept incorrect Authorization headers for development purposes I think.

Reply:
Makes sense!